main/uart_nic.c

```diff
@@ -151,24 +191,106 @@ static void IRAM_ATTR send_link_status(uint8_t up) {
     }
 }
 
-static void IRAM_ATTR probe_task(void* arg) {
-    wifi_scan_config_t config;
+static void IRAM_ATTR do_wifi_scan(void * arg) {
+    wifi_scan_config_t config = {0};
```

Comment:
Actually, zeroes are implicit (at least in the C version), so to set something to all-zeroes, one can do just `= {}`.

(Multiple instances through the file)

Reply:
I know I just prefer to be explicit.

Reply:
Looks a bit weird, as the struct has more than one zeroable field 😇. But OK, I'll leave it up to you.